Add summary at the end of Xcode report

mrvincenzo opened the pull request:
Add a report summary when using the `--output-xcode` option.  The report summary will be printed when the muter has finished doing its magic similar to .json and .plaintext 

The xcode report summary prints the following:

1. globalMutationScore
2. totalAppliedMutationOperators
3. numberOfKilledMutants

mrvincenzo commented:
Some tests failed. I will fix what needs fixin' and reopen the pull request

mrvincenzo commented:
1. Unit test passed
2. Acceptance tests passed
3. Regression tests failed both BEFORE and AFTER my changes (see errors below)

```
Test Case '-[muterRegressionTests.RegressionTests muter test report output for bon mot, does not contain any changes]' started.
/Users/ik216a/projects/muter/RegressionTests/RegressionTests.swift:22: error: -[muterRegressionTests.RegressionTests muter test report output for bon mot, does not contain any changes] : failed - Snapshot does not match reference.

@−
"/Users/ik216a/projects/muter/RegressionTests/__Snapshots__/RegressionTests/runRegressionTest-forFixtureNamed-withResultAt.bonmot.json"
@+
"/var/folders/j8/mxnpk_mn3kd4yvytdcjt601452lppt/T/RegressionTests/runRegressionTest-forFixtureNamed-withResultAt.bonmot.json"

@@ −800,5 +800,4 @@
   "globalMutationScore" : 12,
   "numberOfKilledMutants" : 7,
   "totalAppliedMutationOperators" : 62
 }
−
Test Case '-[muterRegressionTests.RegressionTests muter test report output for bon mot, does not contain any changes]' failed (0.067 seconds).
```

rakaramos commented:
hey @mrvincenzo, thanks for contributing and sorry for making you wait, lately things are sorta crazy down here!

Does this shows before? It seems like we need to update the regression tests.
```diff
@@ −800,5 +800,4 @@
   "globalMutationScore" : 12,
   "numberOfKilledMutants" : 7,
   "totalAppliedMutationOperators" : 62
 }
```

Also, would mind adding a couple of unit tests for the footer only feature?

mrvincenzo commented:
hi @rakaramos. 
> Does this shows before? It seems like we need to update the regression tests.

Yes, these tests failed when I ran the regression tests on the master branch before creating the pull request. 


> Also, would mind adding a couple of unit tests for the footer only feature?

Sure, I will.

mrvincenzo commented:
```

@@ −299,135 +299,8 @@
         }
       ],
       "fileName" : "Platform.swift",
       "mutationScore" : 0
−    },
−    {
−      "appliedOperators" : [
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 21,
−              "line" : 30,
−              "utf8Offset" : 713
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 33,
−              "line" : 30,
−              "utf8Offset" : 725
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 21,
−              "line" : 88,
−              "utf8Offset" : 3245
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 33,
−              "line" : 88,
−              "utf8Offset" : 3257
−            }
−          },
−          "testSuiteOutcome" : "buildError"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 21,
−              "line" : 105,
−              "utf8Offset" : 3738
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 21,
−              "line" : 108,
−              "utf8Offset" : 3825
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 29,
−              "line" : 230,
−              "utf8Offset" : 8511
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 41,
−              "line" : 230,
−              "utf8Offset" : 8523
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 29,
−              "line" : 255,
−              "utf8Offset" : 9730
−            }
−          },
−          "testSuiteOutcome" : "failed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 40,
−              "line" : 255,
−              "utf8Offset" : 9741
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        },
−        {
−          "mutationPoint" : {
−            "mutationOperatorId" : "ChangeLogicalConnector",
−            "position" : {
−              "column" : 37,
−              "line" : 280,
−              "utf8Offset" : 11203
−            }
−          },
−          "testSuiteOutcome" : "passed"
−        }
−      ],
−      "fileName" : "StringStyle+Part.swift",
−      "mutationScore" : 10
     },
     {
       "appliedOperators" : [
         {
@@ −797,8 +670,7 @@
       "mutationScore" : 0
     }
   ],
   "globalMutationScore" : 12,
−  "numberOfKilledMutants" : 7,
−  "totalAppliedMutationOperators" : 62
+  "numberOfKilledMutants" : 6,
+  "totalAppliedMutationOperators" : 51
 }
−
Test Case '-[muterRegressionTests.RegressionTests muter test report output for bon mot, does not contain any changes]' failed (0.077 seconds).
```
This is the output of the failed regression test on master (not as I posted originally). It seems that 11 mutation operators were not applied compared to the snapshot.

mrvincenzo commented:
All the failures are coming from `#if os(iOS)` statements in `StringStyle+Part.swift`

mrvincenzo commented:
I figured it out. 
- These [errors](https://github.com/muter-mutation-testing/muter/pull/157#issuecomment-598671617) occur when running the regression tests with Xcode 10 build tools
- These [errors](https://github.com/muter-mutation-testing/muter/pull/157#issuecomment-614022392) occur when running the regression tests with Xcode 11 build tools

The former errors (Xcode 10) can be fixed by removing the extra trailing newline at the end of `runRegressionTest-forFixtureNamed-withResultAt.bonmot.json` snapshot file. Commit is on the way.

rakaramos commented:
That's great @mrvincenzo, thanks for digging into it!
Lemme me know if you need any help!

mrvincenzo commented:
@rakaramos I've updated the Reporter unit tests. I also fixed the Xcode 10 regression tests issue. IMHO PR is ready to be merged.
As for the Xcode 11 issue, it is not related to this PR. I will try to take a look at it later.

mrvincenzo commented:
Bump
